backend/src/test/java/com/google/sps/EmailNotifierTest.java

```diff
@@ -0,0 +1,177 @@
+    users = mock(Gmail.Users.class);
+    // Mock to return list of messages.
+    messages = mock(Gmail.Users.Messages.class);
+    send = mock(Gmail.Users.Messages.Send.class);
+    MimeMessage mimemessage = createTestMessage();
+    testMessage = createMessageFromMime(mimemessage);
+    // Test Instance of email notifier class.
+    emailNotifier = new EmailNotifier("John", "jdoe@gmail.com ", gmail);
+    // Emulating real behavior due to when method, will return list of users.
+    when(gmail.users()).thenReturn(users);
+    // Emulating real behaviour due to when method, will return list of messages
+    when(users.messages()).thenReturn(messages);
+    //    Emulating sending messages, when sent will return send mock.
+    when(messages.send(any(), any())).thenReturn(send);
+    //    When send mock is executed will return test message.
+    when(send.execute()).thenReturn(testMessage);
+  }
+
+  public MimeMessage createTestMessage() throws MessagingException {
+    Properties properties = System.getProperties();
+    Session session = Session.getDefaultInstance(properties);
+    MimeMessage message = new MimeMessage(session);
+    Address[] addresses =
+        new Address[] {
+          new InternetAddress("tcwang@google.com"),
+          new InternetAddress("grantjustice@google.com"),
+          new InternetAddress("kevinhowald@google.com")
+        };
+    message.setFrom(new InternetAddress("Adlib-Step@gmail.com"));
+    message.setSubject("Test Subject");
+    message.setText("Test Text");
+    message.setRecipients(javax.mail.Message.RecipientType.TO, addresses);
+    return message;
+  }
+
+  public Message createMessageFromMime(MimeMessage mimeMessage)
+      throws IOException, MessagingException {
+    ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+    mimeMessage.writeTo(buffer);
+    byte[] bytes = buffer.toByteArray();
+    String encodedEmail = Base64.encodeBase64URLSafeString(bytes);
+    Message message = new Message();
+    message.setRaw(encodedEmail);
+    return message;
+  }
+
+  public MimeMessage convertToMimeMessage(Message msg) throws MessagingException {
+    Properties properties = System.getProperties();
+    Session session = Session.getDefaultInstance(properties);
+    byte[] bytes = msg.decodeRaw();
+    InputStream targetStream = new ByteArrayInputStream(bytes);
+    return new MimeMessage(session, targetStream);
+  }
+
+  @Test
+  public void testMessageHasCorrectApplicationName()
+      throws MessagingException, IOException, GeneralSecurityException {
+    ArgumentCaptor<Message> argument = ArgumentCaptor.forClass(Message.class);
+    emailNotifier.notifyUser();
+    verify(messages).send(any(), argument.capture());
+    String applicationName = convertToMimeMessage(argument.getValue()).getFrom()[0].toString();
+    Truth.assertWithMessage("These two strings should be the same")
+        .that("Adlib-Step@gmail.com")
+        .isEqualTo(applicationName);
```

Comment:
looks like expected and actual are flipped here and below

`assertThat(actual).isEqualTo(expected)`

Since "Adlib-Step@gmail.com" is the expected value, it should go second.

Reply:
Swapped tests expected and actual